src/window/windows/castlewindow_winapi.inc

```diff
@@ -326,6 +380,8 @@ begin
     SetPixelFormat_WGLChoose also get here (as they use our
     Application.wndClassName). They don't set Get/SetWindowLongPtr, so Window here
     will be @nil in this case. }
+  if uMsg = WM_DROPFILES then
```

Comment:
You added this to `WndProc`. This will work, but there's an easier and better way: add handling of `WM_DROPFILES` to `TCastleWindow.WindowProc`. There's a big `case uMsg of` there, where you can add handling of `WM_DROPFILES`. 

This is better, as we already handle most message in `TCastleWindow.WindowProc`, and we check is `Window` valid along the way.

So, please move implementation of this `TCastleWindow.WindowProc`.